VAMobile/src/components/PhotoUpload.tsx

```diff
@@ -0,0 +1,171 @@
+
+import { NAMESPACE } from 'constants/namespaces'
+import { VAIcon } from './index'
+import { themeFn } from 'utils/theme'
+import { useDestructiveActionSheet, useShowActionSheet } from 'utils/hooks'
+import AsyncStorage from '@react-native-async-storage/async-storage'
+import Box, { BoxProps } from './Box'
+import TextView from './TextView'
+import theme from 'styles/themes/standardTheme'
+
+type PhotoUploadProps = {
+  /** width of the photo */
+  width: number
+  /** height of the photo */
+  height: number
+  /** imagePickerResponse with asset to style for component and fileSize */
+}
+
+type StyledImageProps = {
+  /** prop to set image width */
+  width: number
+  /** prop to set image height */
+  height: number
+  /** Hardcoded radius of 50 due to circular design plan */
+  borderRadius: number
+  /** Hardcoded radius of 2 due to design plan */
+  borderWidth: number
+  /** Hardcoded border color of white */
+  borderColor: string
+}
+
+const StyledImage = styled(Image)<StyledImageProps>`
+  width: ${themeFn<StyledImageProps>((_theme, props) => props.width)}px;
+  height: ${themeFn<StyledImageProps>((_theme, props) => props.height)}px;
+  border-radius: ${themeFn<StyledImageProps>((_theme, props) => props.borderRadius)}px;
+  border-width: ${themeFn<StyledImageProps>((_theme, props) => props.borderWidth)}px;
+  border-color: ${themeFn<StyledImageProps>((_theme, props) => props.borderColor)};
+`
+//TODO: Add this back to the VeteranStatusScreen when decided upon
+const PhotoUpload: FC<PhotoUploadProps> = ({ width, height }) => {
+  const { t } = useTranslation(NAMESPACE.COMMON)
+  const confirmAlert = useDestructiveActionSheet()
+  const photoUploadBorderRadius = 50
+  const photoUploadBorderWidth = 2
+  const showActionSheetWithOptions = useShowActionSheet()
+  const VETERAN_STATUS_PHOTO = '@store_veteran_status_photo'
+  const [uri, setUri] = useState('')
+  const options = [t('fileUpload.camera'), t('fileUpload.photoGallery'), t('cancel')]
+  const uploadBorderColor = theme.colors.border.photoUpload
+
+  const getPhotoFromStorage = async (): Promise<void> => {
+    const override = await AsyncStorage.getItem(VETERAN_STATUS_PHOTO)
+    if (override !== null) {
+      setUri(override)
+    }
+  }
+
+  useEffect(() => {
+    if (uri === '') {
+      getPhotoFromStorage()
+    }
+  }, [uri])
+
+  const photo = (): ReactNode => {
+    return (
+      <StyledImage source={{ uri }} width={width} height={height} borderRadius={photoUploadBorderRadius} borderWidth={photoUploadBorderWidth} borderColor={uploadBorderColor} />
+    )
+  }
+
+  const uploadCallback = (response: ImagePickerResponse): void => {
+    const { assets, errorMessage, didCancel } = response
+    if (didCancel) {
+      return
+    } else if (errorMessage) {
+      //TODO Snackbar in this case? Discuss error cases
+      return
+    } else {
+      if (assets?.length === 0) {
+        return
+      }
+      if (assets && assets[0] && assets[0].uri) {
```

Comment:
```suggestion
      if (assets[0].uri) {
```
Since the previous `if` block confirms that `assets[0]` exists if this block is reached, we should only need the last part of this conditional

Reply:
Just saw that Chris has a similar [suggestion](https://github.com/department-of-veterans-affairs/va-mobile-app/pull/6523#discussion_r1300620142) about this block as well

Reply:
Can be simplified to `if (assets?.[0]?.uri)` if you're keeping this.

Even better, it seems like you're referencing `assets[0].uri` several times throughout this block. Might be better to set it as a var above and reference it to make it clearer.

Reply:
replaced with a var and used Narin's suggestion.